lms/static/js/ccx/view/schedule_date_button.js

```diff
@@ -0,0 +1,107 @@
+var define = window.define || RequireJS.define;  // jshint ignore:line
+
+define(
+  [
+    'backbone',
+    'jquery',
+    'underscore',
+    'gettext',
+    'text!templates/ccx/underscore/date-button.underscore'
+  ],
+  function (
+    Backbone,
+    $,
+    _,
+    gettext,
+    scheduleTreeDateButtonTemplate
+  ) {
+    'use strict';
+     return Backbone.View.extend({
+
+       initialize: function(options) {
+        this.dateType = options.dateType;
+        this.date = options.date;
+       },
+
+       render: function () {
+         this.$el.html(_.template(scheduleTreeDateButtonTemplate)({
+           date: this.date
+         }));
+         this.location = this.$el.data("location");
+         var $dateChangeButton = $(this.$el.find("button"));
+
+         $dateChangeButton.attr('href', '#enter-date-modal').
+          leanModal({closeButton: '.close-modal'});
+         $dateChangeButton.click(this.editDateCCXSchedule());
+
+         return this;
+       },
+
+       editDateCCXSchedule: function() {
+         var self = this;
+         return function () {
+           var dateParts;
+           var row = $(this).closest('tr');
+           var modal = $('#enter-date-modal')
+             .data('what', self.dateType)
+             .data('location', row.data('location'));
+
+           modal.find('h2').text(
+             self.dateType === 'due' ? gettext("Enter Due Date and Time") :
+               gettext("Enter Start Date and Time")
+           );
+
+           if (self.date) {
+             dateParts = self.date ? self.date.split(' ') : ['', ''];
+             var date = dateParts[0];
+             var time = dateParts[1];
+
+             modal.find('input[name=date]').val(date);
+             modal.find('input[name=time]').val(time);
+           }
+
+           modal.focus();
+           $(document).on('focusin', function (event) {
+             try {
+               if (!_.isUndefined(event.target.closest('.modal').id) &&
+                 event.target.closest('.modal').id !== 'enter-date-modal' &&
+                 event.target.id !== 'enter-date-modal') {
+                 event.preventDefault();
+                 modal.find('.close-modal').focus();
+               }
+             } catch (err) {
+               event.preventDefault();
+               modal.find('.close-modal').focus();
+             }
+           });
+           modal.find('.close-modal').click(function () {
+             $(document).off('focusin');
+           });
+
+           modal.find('form').off('submit').on('submit', function(event) {
+             event.preventDefault();
+             var date = $(this).find('input[name=date]').val(),
+             time = $(this).find('input[name=time]').val();
+             var valid_date = new Date(date);
+             if (isNaN(valid_date.valueOf())) {
```

Comment:
It would be nicer if the error message was shown directly in the modal, instead of opening another dialog.

If this feature was going to be widely used, I would go further to recommend that the submit button is disabled until the data validates properly (with display of validation error messages).